python/lsst/ci/prepare.py

```diff
@@ -26,6 +29,47 @@
 ASYNC_QUEUE_WORKERS = 8
 
 
+class AgentManager:
+    """Handles agent label determination and verification."""
+
+    def __init__(self):
+        self.agent = self.verify_agent()
```

Comment:
Seems maybe a bit much to go through a class and three levels of functions to do this when it's unlikely that the inner methods will be called separately.